uvtools/plot.py

```diff
@@ -723,3 +723,185 @@ def plot_diff_uv(uvd1, uvd2, pol=None, check_metadata=True, bins=50):
+
+    Notes
+    -----
+    This function extracts the visibility waterfall corresponding to the 
+    provided antpairpol and flattens it by taking the mean along the axis 
+    not being used. For true 1d arrays this just modifies the shape, but it 
+    might not exactly be the desired behavior--so be warned.
+    """
+    if check_metadata:
+        utils.check_uvd_pair_metadata(uvd1, uvd2)
+
+    if plot_type not in ("base", "dual", "both"):
+        raise ValueError(
+            "You must specify whether to make one or two plots with "
+            "the ``plot_type`` parameter. You may choose to plot the "
+            "visibility difference as a function of frequency/time by "
+            "setting ``plot_type`` to 'base', or you can choose to "
+            "plot the difference in Fourier space by setting "
+            "``plot_type`` to 'dual'. If you would like to plot both, "
+            "then set ``plot_type`` to 'both'."
+        )
+
+    dimensions_to_duals = {"time" : "fr", "freq" : "dly"} 
+
+    if dimension is None:
+        dimension = "time" if uvd1.Ntimes > uvd1.Nfreqs else "freq"
+        if uvd1.Ntimes == uvd1.Nfreqs:
+            warnings.warn(
+                "The UVData objects passed have the same number of " \
+                "times as they do frequencies. You did not specify " \
+                "which dimension to use, so the difference plots " \
+                "will be made along the time axis."
+            )
+
+    if dimension not in ("time", "freq"):
+        raise ValueError(
+            "You must specify whether the visibilities are a function "
+            "of time or frequency by setting the ``dimension`` "
+            "parameter to 'time' or 'freq', respectively."
+        )
+    
+    dual = dimensions_to_duals[dimension]
+
+    use_axis = 0 if dimension == "time" else 1
+    proj_axis = (use_axis + 1) % 2
+
+    # get visibility data
+    vis1 = uvd1.get_data(antpairpol).mean(axis=proj_axis)
+    vis2 = uvd2.get_data(antpairpol).mean(axis=proj_axis)
+
+    # use same approach as in plot_diff_waterfall
+    # get important metadata
+    times = np.unique(uvd1.time_array) # days
+    lsts = np.unique(uvd1.lst_array) # radians
+    freqs = np.unique(uvd1.freq_array) # Hz
+
+    # import astropy for unit conversions
+    import astropy.units as u
+    frs = utils.fourier_freqs(times * u.day.to('s')) # Hz
+    dlys = utils.fourier_freqs(freqs) # s
+
+    # make dictionary of plotting parameters
+    plot_params = {"time" : lsts, # radians
+                   "freq" : freqs / 1e6, # MHz
+                   "fr" : frs * 1e3, # mHz
+                   "dly" : dlys * 1e9 # ns
+                   }
+
+    # now do the same for abscissa labels
+    labels = {"time" : "LST [radians]",
+              "freq" : "Frequency [MHz]",
+              "fr" : "Fringe Rate [mHz]",
+              "dly" : "Delay [ns]"
+              }
+
+    # and now for ordinate labels
+    vis_labels = {"time" : r"$V(t)$ [Jy]",
```

Comment:
Are the visibility units in the `UVData` objects guaranteed to be Jy? Can sometimes be mK.

Reply:
They are not; I've updated the code to use the visibility units from `uvd1`.